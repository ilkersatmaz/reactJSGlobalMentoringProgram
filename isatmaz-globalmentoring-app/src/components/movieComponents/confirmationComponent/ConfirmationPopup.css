.confirmation-form{
    padding: 60px;
    background-color: #232323;
}

.confirmation-form-logo{
    width: 50px;
    height: 50px;
}
.confirmation-form-header{
    font-size: 40px;
    color:#FFFFFF;
    margin: 0;
}
.confirmation-form-div{
    display: flex;
    flex-direction: column;
    column-gap: 50px;
    row-gap: 30px;
    align-items: center;
}

.confirmation-text{
    color: #FFFFFF;
    font-size: 20px;
}

.confirmation-form-close{
    width: 25px;
    height: 25px;
    position: absolute;
    top: 0;
    right: 0;
    margin: 10px;
    color: #FFFFFF;
}