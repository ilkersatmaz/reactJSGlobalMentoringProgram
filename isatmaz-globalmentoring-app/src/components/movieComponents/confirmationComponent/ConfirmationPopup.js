import React from 'react';

import './ConfirmationPopup.css';
import { BsCheckCircleFill } from 'react-icons/bs';
import {AiOutlineClose} from 'react-icons/ai';

class ConfirmationPopup extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState({value: event.target.value});
  }

  handleSubmit(event) {
    alert('movie was removed from the list: ' + this.state.value);
    event.preventDefault();
  }

  render() {
    return (
      <form className='confirmation-form' onSubmit={this.handleSubmit}>        
        <div className='confirmation-form-div'>
          <AiOutlineClose className='confirmation-form-close' onClick={()=>{this.popup.destroy()}}/>
          <BsCheckCircleFill fill='#F65261' stroke='#F65261' className='confirmation-form-logo'/>
          <p className='confirmation-form-header'> CONGRATULATIONS !</p>
          <p className='confirmation-text'>
            The movie has been added to database successfully
          </p>      
        </div>        
       
      </form>
    );
  }
}

export default ConfirmationPopup