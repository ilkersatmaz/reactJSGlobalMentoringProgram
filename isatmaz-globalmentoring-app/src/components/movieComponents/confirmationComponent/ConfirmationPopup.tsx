import React from 'react';

import './ConfirmationPopup.css';
import { BsCheckCircleFill } from 'react-icons/bs';
import {AiOutlineClose} from 'react-icons/ai';

function ConfirmationPopup (){

  const handleSubmit = (event: any) => {
    event.preventDefault();
  }
  return (
    <form className='confirmation-form' onSubmit={handleSubmit}>        
      <div className='confirmation-form-div'>
        <AiOutlineClose className='confirmation-form-close' onClick={()=>{}}/>
        <BsCheckCircleFill fill='#F65261' stroke='#F65261' className='confirmation-form-logo'/>
        <p className='confirmation-form-header'> CONGRATULATIONS !</p>
        <p className='confirmation-text'>
          The movie has been added to database successfully
        </p>      
      </div>        

    </form>
  );
}

export default ConfirmationPopup