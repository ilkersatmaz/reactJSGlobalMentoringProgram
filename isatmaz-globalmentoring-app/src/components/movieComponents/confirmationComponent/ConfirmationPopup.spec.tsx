import { render } from '@testing-library/react';
import React from 'react';
import ConfirmationPopup from './ConfirmationPopup';

describe('Confirmation Popup component', () => {
  it("should match the snapshot", () => {  
    const setIsConfirmationOpen=jest.fn();
    const { asFragment } = render(<ConfirmationPopup setIsConfirmationOpen={setIsConfirmationOpen}/>);
    
    expect(asFragment).toMatchSnapshot();
  });
})