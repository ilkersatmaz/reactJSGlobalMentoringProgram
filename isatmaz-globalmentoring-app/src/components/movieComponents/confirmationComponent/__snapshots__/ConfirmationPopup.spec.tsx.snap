// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Popup component should match the snapshot 1`] = `[Function]`;
